src/runtime_src/core/pcie/driver/linux/xocl/subdev/xmc.c

```diff
@@ -2577,28 +2600,66 @@ QSFP_BIN_ATTR(3);
 	&bin_attr_qsfp##PORT##_upper_page2, \
 	&bin_attr_qsfp##PORT##_upper_page3 \
 
-#define QSFP_IO_CONFIG(PORT) \
+/* qsfp_control, e.g. qsfp0_control */
+#define QSFP_CONTROL(PORT) 							\
+static ssize_t qsfp##PORT##_control_write(                                      \
+	struct file *filp, struct kobject *kobj, 	                        \
+	struct bin_attribute *attr, char *buffer, loff_t off, size_t count)     \
+{                                                                               \
+	struct xocl_xmc *xmc =                                                  \
+		dev_get_drvdata(container_of(kobj, struct device, kobj));       \
+	return xmc_qsfp_write(xmc, PORT, buffer, off, count);                   \
+}
+
+QSFP_CONTROL(0)
+QSFP_CONTROL(1)
+QSFP_CONTROL(2)
+QSFP_CONTROL(3)
+
+static BIN_ATTR_WO(qsfp0_control, 0);
```

Comment:
is sysfs nodes created by BIN_ATTR_WO() allows sudo access or do you have to su to root to write to it?

Reply:
I suppose those are defined by the linux API, https://elixir.bootlin.com/linux/latest/source/include/linux/sysfs.h#L227
aka. RW as 0644, RO as 0444, WO as 02000. 

If we should customize the permission. We will need to not use BIN_ATTR_WO macro, instead we can define the bin_attribute directly as the same as cmc_image_write

Reply:
02000 means that only root can write to it. but i'd think that we would like to allow sudoers to write to it, right? so we can't use BIN_ATTR_WO here?

Reply:
adjust RW to 0600 which is 2(W)+4(R)
adjust W to 0200 which is 2(W)

verified sudoers works.